source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst

```diff
@@ -716,6 +819,29 @@ Why is waitQueueTimeoutMS optional for some drivers?
 
 We are anticipating eventually introducing a single client-side timeout mechanism, making us hesitant to introduce another granular timeout control. Therefore, if a driver/language already has an idiomatic way to implement their timeouts, they should leverage that mechanism over implementing waitQueueTimeoutMS.
 
+Why must ensuring minPoolSize require the use of a background thread or async I/O?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Without the use of a background thread, minPoolSize is ensured during
+checkOut. `Connections <#connection>`_ may not be checked into the pool before
+being established, however, so establishment also happens during
+checkOut. If `Connection <#connection>`_ establishment were done in a blocking
+fashion, the first operations after a clearing of the pool would
+experience unacceptably high latency, especially for larger values of
+minPoolSize. Thus, minPoolSize either needs to be ensured via a
+background thread (which is acceptable to block) or via the usage of
+non-blocking (async) I/O.
+
+Why must closing a connection be non-blocking?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Because idle and perished `Connections <#connection>`_ are cleaned up as part of
+checkOut, performing blocking I/O while closing such `Connections <#connection>`_
+would block application threads, introducing unnecessary latency. Once
+a `Connection <#connection>`_ is marked as "closed", it will not be checked out
+again, so ensuring the socket is torn down does not need to happen
+immediately and can happen at a later time, either via async I/O or a
+background thread. 
```

Comment:
I agree with this rationale, however I think that calling `close` on a socket is actually already asynchronous, so that the drivers do not necessarily need to call close in a background thread.

Again it may be  a good idea to cover asynchronous close in a separate DRIVERS ticket.

Reply:
Sounds good to me. I updated the rationale title to be "Why should closing ..." so I could keep it in here for now, is that okay? I'll file a DRIVERS ticket once this PR is merged.

Reply:
:+1: